### PR TITLE
Added possibiity to add a demo disclaimer banner

### PR DESCRIPTION
**Pull request by ChristophNiehoff**

### CODE REVIEW

File: src/client/components/banner/banner.css [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  font-size: 115%;
+  font-weight: 900;
+  left: -8rem;
+  max-width: 31rem;
+  min-width: 31rem;
+  padding: 7rem;
+  position: fixed;
+  text-align: center;
+  top: 14rem;
+  -webkit-transform: rotate(-45deg);
```
Comment (ghost91-) [inline src/client/components/banner/banner.css:13]: Is there a reason for specifying these vendor properties? I thought `transform` was pretty well supported by now, so using it should be fine?

File: src/client/components/banner/banner.test.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import '@testing-library/jest-dom';
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import Banner from './banner';
+
+describe('Banner', () => {
+  it('should render link if env var is defined', async () => {
+    // given
+    process.env.REACT_APP_EOP_BANNER_TEXT = 'This is a banner text'
```
Comment (ghost91-) [inline src/client/components/banner/banner.test.tsx:9]: Thinking about it, we should probably cleanup `process.env` after each test. Something like
```ts
function withCleanupEnv() {
  const env = process.env;

  beforeEach(() => {
    process.env = { ...process.env };
  });
  
  afterEach(() => {
    process.env = env;
  });
}
```
That can then be used in all tests that mess with `process.env`.

File: src/client/components/banner/banner.test.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    render(<Banner />);
+
+    // when
+    const banners = await screen.queryAllByText('This is a banner text');
```
Comment (ghost91-) [inline src/client/components/banner/banner.test.tsx:13]:
```suggestion
    const banners = screen.queryAllByText('This is a banner text');
```
`query...` queries are not async.

However, I would suggest to use `find...` queries (which are async) instead, where possible:
```tsx
// when
const banner = await screen.findByText('This is a banner text');

// then
expect(banner).toBeInTheDocument();
```

Also applies to the other PR.

File: src/client/components/banner/banner.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import './banner.css';
+
+
+const Banner: FC = () => {
+  if (process.env.REACT_APP_EOP_BANNER_TEXT) {
+    return (
+      <div className='banner'>{process.env.REACT_APP_EOP_BANNER_TEXT}</div>
+      );
+  }
+  return (<></>);
```
Comment (ghost91-) [inline src/client/components/banner/banner.tsx:12]:
```suggestion
  return null;
```

**Review by ghost91- — CHANGES_REQUESTED** [review]
One general observation: Since the banner and imprint and privacy notice are all included in all pages, wouldn't it make sense to have a layout component, for all pages? Then we only have to use the layout component instead of including these three things on all pages.

File: docker/client.dockerfile [hunk 5]
```diff
@@ -11,6 +12,7 @@ COPY ./.eslintrc.cjs ./.eslintrc.cjs
 COPY ./.prettierignore ./.prettierignore
 COPY ./.prettierrc.cjs ./.prettierc.cjs
 COPY ./src ./src
+ENV REACT_APP_EOP_BANNER_TEXT=$EOP_BANNER_TEXT
```
[the remark this answers is not in the document]
Comment (ChristophNiehoff) [reply docker/client.dockerfile:15]: Done

File: src/client/components/banner/banner.test.tsx (hunk 2 above)
Comment (ChristophNiehoff) [reply src/client/components/banner/banner.test.tsx:9]: done

File: src/client/components/banner/banner.tsx (hunk 4 above)
Comment (ChristophNiehoff) [reply src/client/components/banner/banner.tsx:12]: done

File: src/client/components/banner/banner.test.tsx (hunk 3 above)
Comment (ChristophNiehoff) [reply src/client/components/banner/banner.test.tsx:13]: done

File: src/client/components/banner/banner.css (hunk 1 above)
Comment (ChristophNiehoff) [reply src/client/components/banner/banner.css:13]: I believe that is what ChatGPT told me ;) I believe we can keep it. Doesn't hurt and maybe someone with a unsupported browser needs it?

**Pull request merged**